### ISSUE TITLE
[Rule] flag as breaking changes, when x-ms-discriminator-value changes

### ISSUE THREAD

**Issue by weidongxu-microsoft** [Azure/openapi-diff#241]
Case: https://github.com/Azure/azure-rest-api-specs/pull/20376#discussion_r959039927

Reason: 
`x-ms-discriminator-value` is used in client to choose the correct type/class to initialize.

Changing its value will break old SDK in runtime (there will be no API change, but old SDK will break in runtime), or client code (if customer uses REST API directly, and use its value to match the type).

**Comment by jianyexi** [comment]
@mikekistler could you help confirm this rule?

**Comment by mikekistler** [comment]
I agree that this should be flagged by the breaking changes tooling. There may be cases where this isn't breaking, but there certainly are cases where it is, so we need to flag it.
